src/bootstrap/src/core/config/config.rs

```diff
@@ -28,6 +28,51 @@ use crate::utils::cache::{INTERNER, Interned};
 use crate::utils::channel::{self, GitInfo};
 use crate::utils::helpers::{self, exe, output, t};
 
+/// Each path in this list is considered "allowed" in the `download-rustc="if-unchanged"` logic.
+/// This means they can be modified and changes to these paths should never trigger a compiler build
+/// when "if-unchanged" is set.
+///
+/// NOTE: Paths must have the ":!" prefix to tell git to ignore changes in those paths during
+/// the diff check.
+///
+/// WARNING: Be cautious when adding paths to this list. If a path that influences the compiler build
+/// is added here, it will cause bootstrap to skip necessary rebuilds, which may lead to risky results.
+/// For example, "src/bootstrap" should never be included in this list as it plays a crucial role in the
+/// final output/compiler, which can be significantly affected by changes made to the bootstrap sources.
+pub(crate) const RUSTC_IF_UNCHANGED_ALLOWED_PATHS: &[&str] = &[
+    ":!.clang-format",
+    ":!.editorconfig",
+    ":!.git-blame-ignore-revs",
+    ":!.gitattributes",
+    ":!.gitignore",
+    ":!.gitmodules",
+    ":!.ignore",
+    ":!.mailmap",
+    ":!CODE_OF_CONDUCT.md",
+    ":!CONTRIBUTING.md",
+    ":!COPYRIGHT",
+    ":!INSTALL.md",
+    ":!LICENSE-APACHE",
+    ":!LICENSE-MIT",
+    ":!LICENSES",
+    ":!README.md",
+    ":!RELEASES.md",
+    ":!REUSE.toml",
+    ":!config.example.toml",
+    ":!rust-bors.toml",
+    ":!rustfmt.toml",
+    ":!tests",
+    ":!triagebot.toml",
+    ":!src/ci/cpu-usage-over-time.py",
+    ":!src/ci/publish_toolstate.sh",
+    ":!src/doc",
+    ":!src/etc",
```

Comment:
There's a bunch of miscellaneous stuff in there. Are you 100% confident none of it affects the build?

As others said before me, the attitude here should be to start with an empty list, and only add things where we are fully confident they are fine to add, and adding them is worth it in terms of caching.

Reply:
Yeah, it’s too risky to include that and definitely not worth caching.

Reply:
Which one doesn’t seem obviously safe to cache from the current list? Because even if they are rarely merged alone with bors, they will still be very useful in PR CI (by cutting one of the runner’s times in half).

Reply:
As several people said above, that is the wrong question. You had several items in your list that turned out to be potentially problematic. Clearly you considered them "obviously safe", and they weren't.

So unless there's some obvious benefit to including a directory -- like, we regularly have PRs that change only those files -- it should not be included.

Reply:
Actually I never considered them "obviously safe". I wasn’t sure about them and was hoping to get some help from the reviews.

But sure, I can reduce the list even more.